matrices.tex

```diff
@@ -25,6 +25,8 @@
 {x - sqrt(y*(x + y)): 1, x + sqrt(y*(x + y)): 1}
 \end{verbatim}
 
+%% TODO a Python list is not a dense representation...
+%% Is this supposed to say flat? - scopatz
```

Comment:
It sounds like you're conflating dense and contiguous. Python lists may not be contiguous, but they are dense. `[0]*1000` takes (amortized) 1000 as much space to store as `[0]` (same with `[None]*1000`). Javascript arrays (from my understanding of them) are an example of a non-dense array. You can have a Javascript array where the first 1000 items are `undefined` and the 1001st item is `1` and it takes up as much space as `[1]`. The closest thing to a non-dense array in Python is a dictionary indexed by non-negative integers (except you'd need a wrapper because all the methods like len and iteration are wrong).  

If I Google "dense data structure" I only get results that are related to matrices or polynomials. If I Google "dense array" I get references to the non-denseness of Javascript arrays.  Based on that I'd say this is a standard usage of the term. 

I'm all for adding clarity. Should we define "dense" and "sparse"?